### PR TITLE
Disable Trex job retries to prevent race conditions

### PR DESCRIPTION
**Pull request by tkrishtop**
In the [daily logs](https://www.distributed-ci.io/jobs/e3690773-551a-45dc-a5ac-af6393a7bda3/jobStates?sort=date), we can observe race conditions occurring between several TRex jobs running in parallel: 

```
2023_09_04_job_logs]$ grep -a trexapp example-cnf_events.log 
3m59s       Normal    TestStarted            trexapp/trex-app1                                                         Started streams with profile (default) at rate (10kpps) for (120)s
41s         Normal    TestStarted            trexapp/trex-app1                                                         Started streams with profile (default) at rate (10kpps) for (120)s
9m49s       Normal    TestStarted            trexapp/trex-app1                                                         Started streams with profile (default) at rate (10kpps) for (120)s
7m24s       Normal    TestCompleted          trexapp/trex-app1                                                         Profile (default) with size (64) with rate (10kpps) for (120)s have completed
5m38s       Normal    TestStarted            trexapp/trex-app1                                                         Started streams with profile (default) at rate (10kpps) for (120)s
2m21s       Normal    TestStarted            trexapp/trex-app1                                                         Started streams with profile (default) at rate (10kpps) for (120)s
7m24s       Normal    TestFailed             trexapp/trex-app1                                                         Test has failed with packets loss of 263181 pkts
```
We have 5 TestStarted events, one TestCompleted and one TestFailed. That means, that only the first test was completed.

We then rerun the test 4 more times. However, every time it considers the test as done because of that event TestCompleted from the first test in the namespace. That means rerunning the test as it is now causes these competing situations. 

To avoid this, we should clean things up by deleting the namespace and reinstalling all the operators. For the moment, I simply deactivated job retries causing race conditions, and will be concentrating on fixing the job itself.

### CODE REVIEW

**Comment by tkrishtop** [comment]
check dallas ocp-4.13-vanilla example-cnf

**Comment by tkrishtop** [comment]
check dallas ocp-4.13-vanilla example-cnf

File: roles/example-cnf-app/tasks/trex/app.yaml
Comment (capolrik) [inline roles/example-cnf-app/tasks/trex/app.yaml:91]: By default, the duration of the burst is 120s, so I think you can increase this wait :)

**Review by capolrik — COMMENTED** [review]
Thank you for the change! I agree with you, we should remove the broken retry logic for now! I'll approve this when the test 'll pass

File: roles/example-cnf-app/tasks/trex/app.yaml
Comment (tkrishtop) [reply roles/example-cnf-app/tasks/trex/app.yaml:91]: Thanks for the info, I think we actually wait for the burst to finish while waiting 300 seconds for the `TestCompleted` event to show up, just one task above.

Here, it is just a minor pause between having the `TestCompleted` event and the `TestFailed`/`TestPassed` event, it should be almost immediate, [there are only some log dumps in-between these two](https://github.com/openshift-kni/example-cnf/blob/eac311b98456edfa5b40a7a8512817962d6b810f/trex-container-app/app/pyfiles/trexstats.py#L111-L138).

Comment (tkrishtop) [reply roles/example-cnf-app/tasks/trex/app.yaml:91]: Your comment made me think that we could write this better, searching for both selectors at once `"reason==TestPassed,reason==TestFailed"` and then using `failed_when` for the check if `TestFailed` is present. I'll wait for the current test to finish, to validate the loop deactivation, and then refine this part.

**Comment by tkrishtop** [comment]
Hi @capolrik thank you again for the review. The change worked as expected, I got rid of the retries and refined the part waiting for the TestPassed/TestFailed event: https://www.distributed-ci.io/jobs/cb3af2db-3892-4406-ad2f-0feaed1ca691/jobStates?sort=date&task=ef2aeb04-e850-44ed-8815-871bbe9826b8

File: roles/example-cnf-app/tasks/retry-trex.yaml
Comment (capolrik) [inline roles/example-cnf-app/tasks/retry-trex.yaml:4]: Can you leave the name as trex-app1 ? I used this name to remove the events that pollutes the results during the upgrade.

Comment (capolrik) [reply roles/example-cnf-app/tasks/retry-trex.yaml:4]: Or I can update the upgrade hook after we merge this, as you wish :) (I agree, it is an hard-coded value)

**Review by capolrik — APPROVED** [review]
Thanks!

File: roles/example-cnf-app/tasks/retry-trex.yaml
Comment (tkrishtop) [reply roles/example-cnf-app/tasks/retry-trex.yaml:4]: we discussed by chat, and we'll go with the hook upgrade. Thank you @capolrik !

**Pull request merged**
